src/overlays/actors/ovl_En_Goma/z_en_goma.c

```diff
@@ -73,77 +104,796 @@ static ColliderCylinderInit D_80A4B7CC = {
+    Math_ApproachZeroF(&this->actor.speedXZ, 0.5f, 2.0f);
+    Math_ApproachS(&this->actor.shape.rot.y, Actor_WorldYawTowardActor(&this->actor, &PLAYER->actor), 2, 3000);
+
+    if (this->actionTimer == 0) {
+        EnGoma_SetupChasePlayer(this);
+    }
+}
+
+void EnGoma_ChasePlayer(EnGoma* this, GlobalContext* globalCtx) {
+    SkelAnime_Update(&this->skelanime);
+
+    if (Animation_OnFrame(&this->skelanime, 1.0f) || Animation_OnFrame(&this->skelanime, 5.0f)) {
+        if (this->actor.params < 6) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOMA_BJR_WALK);
+        } else {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOMA_JR_WALK);
+        }
+    }
+
+    Math_ApproachF(&this->actor.speedXZ, 3.3333333f, 0.5f, 2.0f);
+    Math_ApproachS(&this->actor.world.rot.y, this->actor.yawTowardsPlayer, 3, 2000);
+    Math_ApproachS(&this->actor.shape.rot.y, this->actor.world.rot.y, 2, 3000);
+
+    if (this->actor.bgCheckFlags & 1) {
+        this->actor.velocity.y = 0.0f;
+    }
+    if (this->actor.xzDistToPlayer <= 150.0f) {
+        EnGoma_SetupPrepareJump(this);
+    }
+}
+
+void EnGoma_SetupStunned(EnGoma* this, GlobalContext* globalCtx) {
+    this->actionFunc = EnGoma_Stunned;
+    this->stunTimer = 100;
+    Animation_MorphToLoop(&this->skelanime, &D_06001548, -5.0f);
+    this->actionTimer = (s16)Rand_ZeroFloat(15.0f) + 3;
+
+    if (this->actor.params < 6) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOMA_BJR_FREEZE);
+    } else {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOMA_JR_FREEZE);
+    }
+}
+
+void EnGoma_Stunned(EnGoma* this, GlobalContext* globalCtx) {
+    Actor_SetColorFilter(&this->actor, 0, 180, 0, 2);
+    this->visualState = 2;
+
+    if (this->actionTimer != 0) {
+        SkelAnime_Update(&this->skelanime);
+    }
+
+    if (this->actor.bgCheckFlags & 1) {
+        this->actor.velocity.y = 0.0f;
+        Math_ApproachZeroF(&this->actor.speedXZ, 0.5f, 2.0f);
+    }
+
+    if (this->stunTimer == 0) {
+        EnGoma_SetupStand(this);
+    } else if (--this->stunTimer < 30) {
+        if (this->stunTimer & 1) {
+            this->actor.world.pos.x += 1.5f;
+            this->actor.world.pos.z += 1.5f;
+        } else {
+            this->actor.world.pos.x -= 1.5f;
+            this->actor.world.pos.z -= 1.5f;
+        }
+    }
+}
+
+void EnGoma_LookAtPlayer(EnGoma* this, GlobalContext* globalCtx) {
+    s16 eyePitch;
+    s16 eyeYaw;
+
+    eyeYaw = Actor_WorldYawTowardActor(&this->actor, &PLAYER->actor) - this->actor.shape.rot.y;
+    eyePitch = Actor_WorldPitchTowardActor(&this->actor, &PLAYER->actor) - this->actor.shape.rot.x;
+
+    if (eyeYaw > 6000) {
+        eyeYaw = 6000;
+    }
+    if (eyeYaw < -6000) {
+        eyeYaw = -6000;
+    }
+    Math_ApproachS(&this->eyeYaw, eyeYaw, 3, 2000);
+    Math_ApproachS(&this->eyePitch, eyePitch, 3, 2000);
+}
+
+void EnGoma_UpdateHit(EnGoma* this, GlobalContext* globalCtx) {
+    static Vec3f sShieldKnockbackVel = { 0, 0, 20 };
```

Comment:
```suggestion
    static Vec3f sShieldKnockbackVel = { 0.0f, 0.0f, 20.0f };
```
Unless it needed to be the other way.